viewing_party/party.py

```diff
@@ -1,23 +1,135 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if not title or not genre or not rating:
+        return None 
+    movie_dict = {}
+    if type(title) == str and type(genre) == str and type(rating) == int or type(rating) == float or int:

```

Comment:
we noticed after we submitted but the or int at the end of the line is not needed. 
```suggestion
    if type(title) == str and type(genre) == str and type(rating) == int or type(rating) == float or int:

```

Reply:
Thanks for pointing this out. Including a check like this (though not required by the project prompt) can be helpful, as it can identify bad data before it gets into our objects. If we didn't have this check, Bad data could sneak into an object at some point, only to cause problems later on. Then we'd have to try to trace back to where the bad data came from.

Intersting, the `or int` literally does nothing here. It might be tempting to read the `type(rating) == float or int` as checking that `rating` is either an int or a float, with the earlier separate check for int making the `or int` redundant, but that's not what's happening. Instead, with the previous rating checks, it's more like this
```py
(type(rating) == int) or (type(rating) == float) or (int)
```
`and` and `or` have lower precedence than the comparison operators, so they act to group together multiple entire comparisons, not multiple values within a comparison. So we have the two checks for the type of rating, anfd then `or int` off by itself. It turns out that if rating is an int or float, the the interpretation of `int` on its own doesn't matter. If the value on the left of `or` is truthy, it doesn't even try to evaluate the value on the right. But if the value on the elft is falsy, it does need to evaluate this. It turns out that all "types" (like `int`) are truthy, so this is like having `or True` stuck on the end. This has the effect of deactivating the float and int type checks, because even if rating is neither a float nor an int, the final `or int` will cause that check to pass.

So even more than not being needed, the `or int` check is causing a problem.

One thing to watch out for in a more general case is that explicit `type()` checks can sometimes be overly strict. It's often preferred to do an `isinstance` check, whcih takes into account type inheritance in Python. In other cases, especially if we're not just storing the value to use later, we might just try to use the value without checking the type at all. If it works, it works. If not, hopefully an error gets raised on the part that's unsupported. This tends to be the most flexible thing to do to remain compatible with python's lack of static type checking.

Statically typed languages like the C language family, Java, and many others, do away with needing to perform checks like this by explicitly declaring the valid types of the parameters as part of the function definition. THe compiler would catch an attempt to pass in the wrong kind of data before even running the code.